pkg/proxy/winkernel/proxier.go

```diff
@@ -0,0 +1,1105 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winkernel
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"reflect"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/helper"
+	apiservice "k8s.io/kubernetes/pkg/api/service"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/proxy"
+	"k8s.io/kubernetes/pkg/proxy/healthcheck"
+	"k8s.io/kubernetes/pkg/util/async"
+)
+
+// KernelCompatTester tests whether the required kernel capabilities are
+// present to run the windows kernel proxier.
+type KernelCompatTester interface {
+	IsCompatible() error
+}
+
+// CanUseWinKernelProxier returns true if we should use the Kernel Proxier
+// instead of the "classic" userspace Proxier.  This is determined by checking
+// the windows kernel version and for the existence of kernel features.
+func CanUseWinKernelProxier(kcompat KernelCompatTester) (bool, error) {
```

Comment:
All this WindowsKernelCompatTester stuff doesn't actually do anything that I can see; it always returns nil for IsCompatible().  Perhaps just remove it?

Reply:
There is a TODO to add this function, since old windows kernel wouldn't support this feature and hence kernel proxy wouldn't work for older versions

Reply:
@madhanrm I'm of the opinion that you should remove the bits that don't do anything for now, and add them back with a new PR when you have implemented them fully.  It keeps the original PR smaller and easier for review, and prevents forgotten bits of code from getting committed that never end up getting used.

Reply:
I have taken care of it, and finished the implementation of this NO-OP method.